### PR TITLE
Add doc comments to Dense methods

### PR DESCRIPTION
**Pull request by kortschak**
This is primarily from text submitted by Antoine-Lizee with modifications by Chih-Wei Chang.

Updates #112, #105, #26,

### CODE REVIEW

**Comment by kortschak** [comment]
PTAL @btracey

File: mat64/dense.go [hunk 1]
```diff
@@ -104,10 +116,16 @@ func (m *Dense) isZero() bool {
 	return m.mat.Stride == 0
 }
 
+// Dims returns the number of rows and columns in the matrix.
 func (m *Dense) Dims() (r, c int) { return m.mat.Rows, m.mat.Cols }
 
+// Caps returns the number of rows and columns in the backing matrix. This
```
Comment (btracey) [inline mat64/dense.go]: There is a trailing "This" in the comment. Not sure if there needs to be more or less.

File: mat64/dense.go [hunk 2]
```diff
@@ -498,6 +569,10 @@ func (m *Dense) Stack(a, b Matrix) {
 	w.Copy(b)
 }
 
+// Augment creates the augmented matrix of a and b, where b is placed in the
+// greater indexed columns
```
Comment (btracey) [inline mat64/dense.go]: Needs punctuation.

File: mat64/dense_arithmetic.go [hunk 3]
```diff
@@ -870,6 +926,10 @@ func (m *Dense) Equals(b Matrix) bool {
 	return true
 }
 
+// EqualsApprox compares the matrices represented by b and the receiver, with
+// tolerance for element-wise equality specified by epsilon
```
Comment (btracey) [inline mat64/dense_arithmetic.go]: Needs punctuation.

**Comment by btracey** [comment]
LGTM with the few nits fixed.

File: mat64/dense.go (hunk 1 above)
Comment (kortschak) [inline mat64/dense.go]: I could say it's missing a '!' and that it's an emphatic "This!", But no. I was going to add an explanation and then decided against it.

**Comment by kortschak** [comment]
@lazywei @antoine-lizee please send PRs to add yourselves to gonum/license A+C.

**Pull request merged**